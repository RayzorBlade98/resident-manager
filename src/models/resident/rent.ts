import MonthYear from '_/extensions/date/month_year.extension';
import { CurrencyInCents } from '_/utils/currency/currency.utils';

/**
 * Object containing information about the rent of a single month
 */
export interface RentInformation {
  /**
   * Month and year this rent information is about
   */
  dueDate: MonthYear;

  /**
   * Rent that has to be paid in this month
   */
  rent: CurrencyInCents;

  /**
   * Incidentals that have to be paid in this month
   */
  incidentals: CurrencyInCents;

  /**
   * Date on which the payment was received
   */
  paymentDate?: Date;

  /**
   * Amount that was paid for this month
   */
  paymentAmount?: CurrencyInCents;
}

/**
 * Enum containg status about the rent payment
 */
export enum PaymentStatus {
  /**
   * All costs where paid
   */
  Paid = 'Paid',

  /**
   * Not enough was paid
   */
  PaidPartially = 'PaidPartially',

  /**
   * Nothing was paid
   */
  Unpaid = 'Unpaid',
}
