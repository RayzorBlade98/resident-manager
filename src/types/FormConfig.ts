import { GroupedFormProps } from '_/components/form/GroupedForm/GroupedForm';
import { FormValidationArguments } from '_/hooks/useFormValidation/useFormValidation';

/**
 * Object containing the config of a grouped form
 */
export type FormConfig<TInput extends object, TGroups extends string> = {
  formValidationConfig: Omit<
  FormValidationArguments<TInput>,
  'onSubmitSuccess'
  >;
  formGroupConfig: Omit<
  GroupedFormProps<TInput, TGroups>,
  'children' | 'formErrors'
  >;
};
