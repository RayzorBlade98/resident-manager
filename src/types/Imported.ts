import Prettify from './Prettify';

/**
 * Type that replaces all date types with the string type
 * representing the result of the import of an object from type `T`
 */
type Imported<T extends object> = Prettify<{
  [key in keyof T]: T[key] extends Date | undefined
    ? string
    : T[key] extends object
      ? Imported<T[key]>
      : T[key];
}>;

export default Imported;
