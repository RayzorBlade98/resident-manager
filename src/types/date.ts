/**
 * Enum containing all months as `string`
 */
export enum Month {
  January = 'Januar',
  Febuary = 'Februar',
  March = 'MÃ¤rz',
  April = 'April',
  May = 'Mai',
  June = 'Juni',
  July = 'Juli',
  August = 'August',
  September = 'September',
  October = 'Oktober',
  November = 'November',
  December = 'Dezember',
}

/**
 * Type representing a year
 */
export type Year = number;

/**
 * Information about a specific month and year
 */
export interface MonthYear {
  month: Month;
  year: Year;
}

/**
 * Type of the string representation of a `MonthYear`
 *
 * Format: "[month] [year]"
 */
export type MonthYearString = `${Month} ${Year}`;

/**
 * Type of the string representation of a date including day, month and year
 *
 * Format: `dd.mm.yyyy`
 */
export type DateString = `${string}.${string}.${string}`;

/**
 * Type of the date string representation returned by the date input field
 *
 * Format: `yyyy-mm-dd`
 */
export type DateInputString = `${string}-${string}-${string}`;

/**
 * Class that provides utility functions for `MonthYear` objects
 */
export abstract class MonthYearUtils {
  /**
   * Adds the specified number of months to the given `MonthYear` object
   * @param monthYear `MonthYear` object to which the months should be added
   * @param addedMonths Number of months that should be added (can be negative)
   * @returns new `MonthYear` object with the adjusted month (and year)
   */
  public static addMonths(monthYear: MonthYear, addedMonths = 1): MonthYear {
    if (addedMonths < 0) {
      // Add negative amount of months == subtract months
      return MonthYearUtils.subtractMonths(monthYear, addedMonths * -1);
    }

    const newMonthYear = { ...monthYear };
    if (newMonthYear.month === Month.December) {
      // Year change
      newMonthYear.month = Month.January;
      newMonthYear.year += 1;
    } else {
      // Add one month to the object
      const months = Object.values(Month);
      const oldMonthIndex = months.findIndex(
        (m: Month) => m === newMonthYear.month,
      );
      newMonthYear.month = months[oldMonthIndex + 1];
    }

    // Call `addMonths` recursive until all months got addded
    if (addedMonths !== 1) {
      return MonthYearUtils.addMonths(newMonthYear, addedMonths - 1);
    }
    return newMonthYear;
  }

  /**
   * Checks if two `MonthYear` objects are equal
   * @param a first `MonthYear` object to compare
   * @param b second `MonthYear` object to compare
   * @returns `true` if both the month and year are the same, else `false`
   */
  public static areEqual(a: MonthYear, b: MonthYear): boolean {
    return a.month === b.month && a.year === b.year;
  }

  /**
   * Compares two `MonthYear` objects
   * @param a First `MonthYear` object to compare
   * @param b Second `MonthYear` object to compare
   * @returns
   *
   * - 0 if both object have the same month and year
   * - <0 if the first object contains a date that's earlier than the second object
   * - \>0 if the first object contains a date that's later than the second object
   */
  public static compare(a: MonthYear, b: MonthYear): number {
    if (a.year !== b.year) return a.year - b.year;

    const months = Object.values(Month);
    const aMonthIndex = months.findIndex((m: Month) => m === a.month);
    const bMonthIndex = months.findIndex((m: Month) => m === b.month);
    return aMonthIndex - bMonthIndex;
  }

  /**
   * Returns a `MonthYear` object that contains information of the current month and year
   * @returns `MonthYear` object for the current date
   */
  public static getCurrentMonthYear(): MonthYear {
    const date: Date = new Date();
    const monthIndex: number = date.getUTCMonth();
    const month: Month = Object.values(Month)[monthIndex] as Month;
    const year: Year = date.getUTCFullYear();

    return {
      month,
      year,
    };
  }

  /**
   * Parses a string representation of a `MonthYear` object to its respective object
   * @param monthYear string representation of a `MonthYear` object that should be parsed
   * @returns `MonthYear` object that contains the information of the specified `string`
   */
  public static parseString(monthYear: MonthYearString): MonthYear {
    const splits = monthYear
      .split(/(\s+)/)
      .filter((s: string) => s.trim().length > 0);
    return {
      month: splits[0] as Month,
      year: Number(splits[1]),
    };
  }

  /**
   * Subtracts the specified number of months from the given `MonthYear` object
   * @param monthYear `MonthYear` object from which the months should be subtraced
   * @param subtractedMonths Number of months that should be subtracted (can be negative)
   * @returns new `MonthYear` object with the adjusted month (and year)
   */
  public static subtractMonths(
    monthYear: MonthYear,
    subtractedMonths = 1,
  ): MonthYear {
    if (subtractedMonths < 0) {
      // Subtract negative amount of months == add months
      return MonthYearUtils.addMonths(monthYear, subtractedMonths * -1);
    }

    const newMonthYear = { ...monthYear };
    if (newMonthYear.month === Month.January) {
      // Year change
      newMonthYear.month = Month.December;
      newMonthYear.year -= 1;
    } else {
      // Subtract one month from the object
      const months = Object.values(Month);
      const oldMonthIndex = months.findIndex(
        (m: Month) => m === newMonthYear.month,
      );
      newMonthYear.month = months[oldMonthIndex - 1];
    }

    // Call `subtractMonths` recursive until all months got subtracted
    if (subtractedMonths !== 1) {
      return MonthYearUtils.subtractMonths(newMonthYear, subtractedMonths - 1);
    }
    return newMonthYear;
  }

  /**
   * Converts a `MonthYear` to its string representation
   * @param monthYear `MonthYear` that should be converted
   * @returns string representation of the specified `MonthYear` object with the format "[month] [year]"
   */
  public static toString(monthYear: MonthYear): MonthYearString {
    return `${monthYear.month} ${monthYear.year}`;
  }

  /**
   * Creates a list of `MonthYear` objects that contains each month
   * between the two specified `MonthYear` objects (inclusive these months)
   * @param start Start of the timespan
   * @param end End of the timespan
   * @returns List of `MonthYear` objects with all months between `start` and `end`
   */
  public static timespan(start: MonthYear, end: MonthYear): MonthYear[] {
    const timespan = [{ ...start }];

    // Whether to add a month each loop or subtract one
    const direction = Math.sign(MonthYearUtils.compare(end, start));

    if (MonthYearUtils.areEqual(start, end)) return timespan;

    for (
      let next = MonthYearUtils.addMonths(start, direction);
      !MonthYearUtils.areEqual(next, end);
      next = MonthYearUtils.addMonths(next, direction)
    ) {
      timespan.push(next);
    }
    timespan.push({ ...end });
    return timespan;
  }
}

/**
 * Class that provides utility functions for dates containing day, month and year.
 */
export abstract class DateUtils {
  /**
   * Converts a date string from `yyyy-mm-dd` to `dd.mm.yyyy` format
   * @param date Date in `yyyy-mm-dd` format
   * @returns Converted date in `dd.mm.yyyy` format
   */
  public static convertDateInputString(date: DateInputString): DateString {
    const splits = date.split('-');
    return `${splits[2]}.${splits[1]}.${splits[0]}`;
  }

  /**
   * Converts a date string from `dd.mm.yyyy` to `yyyy-mm-dd` format
   * @param date Date in `dd.mm.yyyy` format
   * @returns Converted date in `yyyy-mm-dd` format
   */
  public static convertToDateInputString(date: DateString): DateInputString {
    const splits = date.split('.');
    return `${splits[2]}-${splits[1]}-${splits[0]}`;
  }

  /**
   * Returns a date string containing the current date
   * @returns Date string containing the current date in `dd.mm.yyyy` format
   */
  public static getCurrentDate(): DateString {
    const date: Date = new Date();
    const day: number = date.getUTCDate();
    const month: number = date.getUTCMonth() + 1;
    const year = String(date.getUTCFullYear());

    const dayPrefix: string = day < 10 ? '0' : '';
    const monthPrefix: string = month < 10 ? '0' : '';

    return `${dayPrefix}${String(day)}.${monthPrefix}${String(month)}.${year}`;
  }
}
