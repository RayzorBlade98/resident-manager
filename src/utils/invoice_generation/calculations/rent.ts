import Invoice from '_/models/invoice/invoice';
import { Resident } from '_/models/resident/resident';

/**
 * Arguments for the rent payment calculation
 */
interface RentPaymentCalculationArguments {
  /**
   * Invoice the rent payment calculation should be added to
   */
  invoice: Invoice;

  /**
   * List of all residents
   */
  residents: Resident[];
}

/**
 * Adds the incidentals calculation to the invoice
 * @param args arguments for the incidentals calculation
 */
export default function addRentPaymentCalculationToInvoice(
  args: RentPaymentCalculationArguments,
): void {
  for (const resident of args.residents) {
    args.invoice.residentInformation[resident.id].rentPayments = resident.rentInformation
      .filter(
        (r) => args.invoice.start <= r.dueDate && r.dueDate <= args.invoice.end,
      )
      .map((r) => {
        const paymentAmount = r.paymentAmount ?? 0;
        return {
          dueDate: r.dueDate,
          rent: r.rent,
          incidentals: r.incidentals,
          paymentAmount,
          paymentMissing: r.rent + r.incidentals - paymentAmount,
        };
      });
  }
}
