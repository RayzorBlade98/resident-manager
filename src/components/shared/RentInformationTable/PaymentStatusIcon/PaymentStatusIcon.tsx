import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import HighlightOffIcon from '@mui/icons-material/HighlightOff';
import { Tooltip } from '@mui/material';
import React from 'react';
import '_/extensions/date/date.extension';
import { convertCurrencyCentsToString, CurrencyInCents } from '../../../../utils/currency/currency.utils';
import RentInformationUtils from '../../../../utils/rent/rent.utils';
import { PaymentStatus, RentInformation } from '_/models/resident/rent';

interface PaymentStatusIconProps {
  /**
   * Rentinformation for which the status should be dispayed
   */
  rentInformation: RentInformation;
}

/**
 * Icon that displays the payment status of a rent information
 */
function PaymentStatusIcon(props: PaymentStatusIconProps): JSX.Element {
  const paymentStatus = RentInformationUtils.getPaymentStatus(
    props.rentInformation,
  );

  const tooltip = (): string => {
    switch (paymentStatus) {
      case PaymentStatus.Paid:
        return `Bezahlt am ${(
          props.rentInformation.paymentDate as Date
        ).toPreferredString()}`;
      case PaymentStatus.Unpaid:
        return 'Unbezahlt';
      case PaymentStatus.PaidPartially:
        return `Teilweise bezahlt am ${(
          props.rentInformation.paymentDate as Date
        ).toPreferredString()} (${convertCurrencyCentsToString(
          props.rentInformation.paymentAmount as CurrencyInCents,
        )} von ${convertCurrencyCentsToString(
          RentInformationUtils.getAmountToPay(props.rentInformation),
        )})`;
      case PaymentStatus.DeductedInInvoice:
        return 'In Nebenkostenrechnung abgerechnet';
      /* istanbul ignore next */
      default:
        throw new Error(`Tooltip for status ${paymentStatus} not implemted`);
    }
  };

  const icon = (): JSX.Element => {
    switch (paymentStatus) {
      case PaymentStatus.Paid:
      case PaymentStatus.DeductedInInvoice:
        return <CheckCircleOutlineIcon color="success" />;
      case PaymentStatus.Unpaid:
        return <HighlightOffIcon color="error" />;
      case PaymentStatus.PaidPartially:
        return <CheckCircleOutlineIcon color="warning" />;
      /* istanbul ignore next */
      default:
        throw new Error(`Icon for status ${paymentStatus} not implemted`);
    }
  };

  return (
    <Tooltip title={tooltip()} arrow>
      {icon()}
    </Tooltip>
  );
}

export default PaymentStatusIcon;
