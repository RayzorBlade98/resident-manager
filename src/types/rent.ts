import { MonthYear, MonthYearUtils } from './date';
import { CurrencyInCents } from '_/utils/currency/currency';

/**
 * Object containing information about the rent of a single month
 */
export interface RentInformation {
  /**
   * Month and year this rent information is about
   */
  dueDate: MonthYear;

  /**
   * Rent that has to be paid in this month
   */
  rent: CurrencyInCents;

  /**
   * Incidentals that have to be paid in this month
   */
  incidentals: CurrencyInCents;

  /**
   * Date on which the payment was received
   */
  paymentDate?: Date;

  /**
   * Amount that was paid for this month
   */
  paymentAmount?: CurrencyInCents;
}

/**
 * Enum containg status about the rent payment
 */
export enum PaymentStatus {
  /**
   * All costs where paid
   */
  Paid = 'Paid',

  /**
   * Not enough was paid
   */
  PaidPartially = 'PaidPartially',

  /**
   * Nothing was paid
   */
  Unpaid = 'Unpaid',
}

/**
 * Class that provides utility functions for `RentInformation` objects
 */
export abstract class RentInformationUtils {
  /**
   * Creates rent information for all months between the due date of the last provided rent information
   * and the current month and adds them to the list of rent information.
   * If the due date of the last provided rent information is in the future, no new information will be added.
   * @param rentInformation List of current rent information
   */
  public static addMissingMonths(rentInformation: RentInformation[]): void {
    const lastRentInformation: RentInformation = rentInformation.at(-1)!; // eslint-disable-line @typescript-eslint/no-non-null-assertion
    if (
      MonthYearUtils.compare(
        lastRentInformation.dueDate,
        MonthYearUtils.getCurrentMonthYear(),
      ) >= 0
    ) {
      // Last rent information isn't in the past
      return;
    }

    const missingRentInformation = RentInformationUtils.timespan(
      MonthYearUtils.addMonths(lastRentInformation.dueDate, 1),
      MonthYearUtils.getCurrentMonthYear(),
      lastRentInformation.rent,
      lastRentInformation.incidentals,
    );
    rentInformation.push(...missingRentInformation);
  }

  /**
   * Returns how much the resident must pay for the specified rent information
   * @param rentInformation `RentInformation` object containing information about all needed payments
   * @returns `amountToPay = rent + incidentals`
   */
  public static getAmountToPay(
    rentInformation: RentInformation,
  ): CurrencyInCents {
    return rentInformation.rent + rentInformation.incidentals;
  }

  /**
   * Returns the status about the payment of the specified rent information
   * @param rentInformation `RentInformation` object containing information about all payments
   * @returns
   *
   * - PaymentStatus.Unpaid: No payment was done
   * - PaymentStatus.Paid: `amountPaid >= amountToPay`
   * - PaymentStatus.PaidPartially: `amountPaid < amountToPay`
   */
  public static getPaymentStatus(
    rentInformation: RentInformation,
  ): PaymentStatus {
    if (!rentInformation.paymentAmount) return PaymentStatus.Unpaid;

    if (
      rentInformation.paymentAmount
      >= RentInformationUtils.getAmountToPay(rentInformation)
    ) return PaymentStatus.Paid;

    return PaymentStatus.PaidPartially;
  }

  /**
   * Creates a list of `RentInformation` objects that contains rent information for each
   * month between the two specified `MonthYear` objects (inclusive these months)
   * @param start Start of the timespan
   * @param end End of the timespan. If `end` < `start` only the start month will be included into the timespan
   * @param rent Amount of rent each object should have
   * @param incidentals Amount of incidentals each object should have
   * @returns List of `RentInformation` objects with rent information for all months between `start` and `end`
   */
  public static timespan(
    start: MonthYear,
    end: MonthYear,
    rent: CurrencyInCents,
    incidentals: CurrencyInCents,
  ): RentInformation[] {
    let timespan = [{ ...start }];

    if (MonthYearUtils.compare(start, end) < 0) {
      // Only create timespan if start is before end
      timespan = MonthYearUtils.timespan(start, end);
    }

    return timespan.map<RentInformation>((m: MonthYear) => ({
      dueDate: m,
      rent,
      incidentals,
    }));
  }
}
