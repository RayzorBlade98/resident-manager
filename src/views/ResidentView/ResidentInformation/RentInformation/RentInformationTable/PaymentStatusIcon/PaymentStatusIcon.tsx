import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import HighlightOffIcon from '@mui/icons-material/HighlightOff';
import { Tooltip } from '@mui/material';
import React from 'react';
import {
  CurrencyInCents,
  convertCurrencyCentsToString,
} from '../../../../../../utils/currency/currency.utils';
import RentInformationUtils from '../../../../../../utils/rent/rent.utils';
import { PaymentStatus, RentInformation } from '_/types/rent';
import '_/extensions/date/date.extension';

interface PaymentStatusIconProps {
  /**
   * Rentinformation for which the status should be dispayed
   */
  rentInformation: RentInformation;
}

/**
 * Icon that displays the payment status of a rent information
 */
function PaymentStatusIcon(props: PaymentStatusIconProps): JSX.Element {
  const paymentStatus = RentInformationUtils.getPaymentStatus(
    props.rentInformation,
  );

  const tooltip = (): string => {
    switch (paymentStatus) {
      case PaymentStatus.Paid:
        return `Bezahlt am ${(
          props.rentInformation.paymentDate as Date
        ).toPreferredString()}`;
      case PaymentStatus.Unpaid:
        return 'Unbezahlt';
      case PaymentStatus.PaidPartially:
        return `Teilweise bezahlt am ${(
          props.rentInformation.paymentDate as Date
        ).toPreferredString()} (${convertCurrencyCentsToString(
          props.rentInformation.paymentAmount as CurrencyInCents,
        )} von ${convertCurrencyCentsToString(
          RentInformationUtils.getAmountToPay(props.rentInformation),
        )})`;
      /* istanbul ignore next */
      default:
        throw new Error(`Tooltip for status ${paymentStatus} not implemted`);
    }
  };

  const icon = (): JSX.Element => {
    switch (paymentStatus) {
      case PaymentStatus.Paid:
        return <CheckCircleOutlineIcon color="success" />;
      case PaymentStatus.Unpaid:
        return <HighlightOffIcon color="error" />;
      case PaymentStatus.PaidPartially:
        return <CheckCircleOutlineIcon color="warning" />;
      /* istanbul ignore next */
      default:
        throw new Error(`Icon for status ${paymentStatus} not implemted`);
    }
  };

  return (
    <Tooltip title={tooltip()} arrow>
      {icon()}
    </Tooltip>
  );
}

export default PaymentStatusIcon;
