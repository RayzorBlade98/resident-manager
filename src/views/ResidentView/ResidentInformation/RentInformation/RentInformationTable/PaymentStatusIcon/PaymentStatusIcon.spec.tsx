import { render } from '@testing-library/react';
import React from 'react';
import PaymentStatusIcon from './PaymentStatusIcon';
import { PaymentStatus, RentInformationUtils } from '_/types/rent';
import RecoilTestWrapper from '_tests/__test_utils__/RecoillTestWrapper';
import RentInformationBuilder from '_tests/__test_utils__/builders/rent_information_builder';

describe('PaymentStatusIcon', () => {
  test.each([
    PaymentStatus.Paid,
    PaymentStatus.Unpaid,
    PaymentStatus.PaidPartially,
  ])('should match snapshot for payment status %s', (status) => {
    // Arrange
    const rentInformation = new RentInformationBuilder()
      .withPayment(1, new Date(2023, 4, 28))
      .build();
    jest
      .spyOn(RentInformationUtils, 'getPaymentStatus')
      .mockReturnValue(status);
    const renderResult = render(
      <RecoilTestWrapper>
        <PaymentStatusIcon rentInformation={rentInformation} />
      </RecoilTestWrapper>,
    );

    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });
});
