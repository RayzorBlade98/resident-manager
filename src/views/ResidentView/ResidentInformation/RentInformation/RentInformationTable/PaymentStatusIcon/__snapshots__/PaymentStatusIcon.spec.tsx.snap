// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentStatusIcon should match snapshot for payment status Paid 1`] = `
<div>
  <svg
    aria-hidden="true"
    aria-label="Bezahlt am 28.05.2023"
    class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
    data-mui-internal-clone-element="true"
    data-testid="CheckCircleOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
</div>
`;

exports[`PaymentStatusIcon should match snapshot for payment status PaidPartially 1`] = `
<div>
  <svg
    aria-hidden="true"
    aria-label="Teilweise bezahlt am 28.05.2023 (0.01€ von 600.00€)"
    class="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
    data-mui-internal-clone-element="true"
    data-testid="CheckCircleOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
</div>
`;

exports[`PaymentStatusIcon should match snapshot for payment status Unpaid 1`] = `
<div>
  <svg
    aria-hidden="true"
    aria-label="Unbezahlt"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
    data-mui-internal-clone-element="true"
    data-testid="HighlightOffIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M14.59 8 12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
    />
  </svg>
</div>
`;
